SmartDeviceLink/private/SDLMenuManager.m

```diff
@@ -654,26 +660,30 @@ - (SDLAddCommand *)sdl_commandForMenuCell:(SDLMenuCell *)cell withArtwork:(BOOL)
     params.menuName = cell.title;
     params.parentID = cell.parentCellId != UINT32_MAX ? @(cell.parentCellId) : nil;
     params.position = @(position);
+    params.tertiaryText = cell.tertiaryText;
+    params.secondaryText = cell.secondaryText;
 
     command.menuParams = params;
     command.vrCommands = (cell.voiceCommands.count == 0) ? nil : cell.voiceCommands;
     command.cmdIcon = (cell.icon && shouldHaveArtwork) ? cell.icon.imageRPC : nil;
     command.cmdID = @(cell.cellId);
+    command.secondaryImage = cell.secondaryArtwork.imageRPC;
+    command.secondaryImage = (cell.secondaryArtwork && shouldHaveArtwork) ? cell.secondaryArtwork.imageRPC : nil;
 
     return command;
 }
 
 - (SDLAddSubMenu *)sdl_subMenuCommandForMenuCell:(SDLMenuCell *)cell withArtwork:(BOOL)shouldHaveArtwork position:(UInt16)position {
     SDLImage *icon = (shouldHaveArtwork && (cell.icon.name != nil)) ? cell.icon.imageRPC : nil;
+    SDLImage *secondaryImage = (shouldHaveArtwork && (cell.secondaryArtwork.name != nil)) ? cell.secondaryArtwork.imageRPC : nil;
```

Comment:
This should also check if the secondary artwork was uploaded successfully before trying to set the `secondaryImage`